### ISSUE TITLE
[Swift] add support for authentications from swagger spec

### ISSUE THREAD

**Issue by cjolif** [swagger-api/swagger-codegen#2493]
Swagger spec does provide the ability to specify the authentication mechanism to be used by a particular API. It seems that Swift generation ignore that part.

**Comment by wing328** [comment]
@cjolif may I know if that's something you're working on? A good starting point is to look at how ObjC API client supports authentication.

**Comment by cjolif** [comment]
@wing328 not in a short term. I do have workaround for my case for now.

**Comment by wing328** [comment]
@cjolif no problem. We'll see who from the community has cycle to contribute this.

For the workaround, do you mean adding default headers?

**Comment by cjolif** [comment]
@wing328, yes my workaround is to add the expected headers as default headers.

**Comment by sergiocg90** [comment]
@wing328 @cjolif could you please post the workaround? I am not able to see it :(

**Comment by wing328** [comment]
@sergiocg90 I believe you need to use the `addHeaders` method to add a header with authentication info (e.g. token):
```
Classes/Swaggers/APIs.swift:    public func addHeaders(aHeaders:[String:String]) {
```
